docs/src/architecture/notes.md

```diff
@@ -1,73 +1,194 @@
+
+</details>
+
+## Note storage mode
+Similar to accounts, there are two storage modes for notes in Miden. Notes can be stored publicly in the [Notes DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#notes-database) with all data visible for everyone. Alternatively, notes can be stored privately committing only the note hash to the Notes DB. 
+
+Every note has a unique single note hash. It is defined as `hash(hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash), vault_hash)`. It is easy to compute the note hash if all the note data is known. To compute a note's hash, we do not need to know the note's `serial_num`. Knowing the hash of the `serial_num` (as well as `script_hash`, `input_hash` and `note_vault`) is also sufficient. We compute the hash of `serial_num` as `hash(serial_num, [0; 4])` to simplify processing within the VM.
+
+Privately stored notes can only be consumed if the note data is known to the consumer. The note data must be provided to the [transaction kernel](../transactions/transaction-kernel.md). That means, there must be some offchain communication to transmit the note's data from the sender to the recipient.
+
+## Note discovery
+Note discovery describes the process of Miden Clients finding notes they want to consume. It will always be possible to send note data directly to the target account, and if the note was previously recored on-chain, it can be consumed. However, it is also possible to querry the Miden Operator and request newly recorded relevant notes. This is done via Note Tags. Tags are part of the Note's metadata and are represented by a `Felt`. 
+
+The `SyncState` API requires the Miden Client to provide a `note_tag` value which is used as a filter in the response. 
+
+## Note consumption
+As with creation, notes can only be consumed in Miden transactions. If a valid transaction consuming an `InputNote` gets verified by the Miden Node, the note's unique nullifier gets added to the [Nullifier DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#nullifier-database) and is therefore consumed. Some notes can be consumed by any account and for some note consumption is restricted.
```

Comment:
Change to by miden txs. 

Change "Node" to "rollup" or "network". 

I would remove the following sentence: Some notes can be consumed by any account and for some note consumption is restricted.

Reply:
> I would remove the following sentence: Some notes can be consumed by any account and for some note consumption is restricted.

I added a new line before that sentence. I though it might be helpful because it introduces the next section "Note recipient to restrict note consumption to certain targets"